src/components/formComponents/worksetsArray.tsx

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import { Workout } from "../../types";
+import { FieldArray } from "formik";
+import {
+  sectionStyle,
+  sectionTitleStyle,
+  RemoveButton,
+  Select,
+  Input,
+  AddButton
+} from ".";
+import {
+  exercisesOptions,
+  intensityTypeOptions,
+  intervalTypeOptions,
+  newWorkset
+} from "../../data";
+
+type WorksetsArrayProps = {
+  values: Workout;
+  workgroupIndex: number;
+  roundIndex: number;
+};
+const WorksetsArray: React.FC<WorksetsArrayProps> = ({
+  values,
+  workgroupIndex,
+  roundIndex
+}) => (
+  <FieldArray
+    name="worksets"
+    render={worksetsArrayHelpers => (
+      <div>
+        {values.workgroups[workgroupIndex].rounds[roundIndex].worksets.map(
+          (_workset, worksetIndex) => (
```

Comment:
Perhaps a personal preference, but given the number of times you reference `workgroups.${workgroupIndex}.rounds.${roundIndex}.worksets.${worksetIndex}` in the fieldNames below, I'd probably do something like:

```tsx
const worksetFieldName = `workgroups.${workgroupIndex}.rounds.${roundIndex}.worksets.${worksetIndex}`

// ... and then:

<Select
  // ...otherProps
  fieldName={`${worksetFieldName}.exercise.name`}
>
```